picard/util/__init__.py

```diff
@@ -175,20 +179,57 @@ def decode_filename(filename):
         return filename.decode(_io_encoding)
 
 
+def _check_windows_min_version(major, build):
+    try:
+        v = sys.getwindowsversion()
+        return v.major >= major and v.build >= build
+    except AttributeError:
+        return False
+
+
+def system_supports_long_paths():
+    """Detects long path support.
+
+    On Windows returns True, only if long path support is enabled in the registry (Windows 10 1607 or later).
+    All other systems return always True.
+    """
+    if not IS_WIN:
+        return True
+    try:
+        # Use cached value
+        return system_supports_long_paths._supported
+    except AttributeError:
+        pass
+    try:
+        # Long path support can be enabled in Windows 10 version 1607 or later
+        if _check_windows_min_version(10, 14393):
+            with winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
+                                r"SYSTEM\CurrentControlSet\Control\FileSystem") as key:
+                supported = winreg.QueryValueEx(key, "LongPathsEnabled")[0] == 1
+        else:
+            supported = False
+        system_supports_long_paths._supported = supported
+        return supported
```

Comment:
I believe these should be `_supported` rather than `supported`,

Reply:
Wrong.  I didn't look close enough.  Sorry about that.

Reply:
No, should be fine that way. `supported`  is the local variable, `system_supports_long_paths._supported`  is the cache value (to avoid the rather expensive check with registry lookup)